File: src/main/java/com/dansoftware/libraryapp/gui/dbcreator/DatabaseCreatorWindow.java
package com.dansoftware.libraryapp.gui.dbcreator;

import com.dansoftware.libraryapp.gui.theme.Theme;
import com.dansoftware.libraryapp.gui.theme.Themeable;
import javafx.scene.Scene;
import javafx.stage.Modality;
import javafx.stage.Stage;
import javafx.stage.Window;

import java.util.List;
import java.util.Objects;

import static com.dansoftware.libraryapp.locale.I18N.getFXMLValues;

/**
 * A DatabaseCreatorWindow is a javaFX {@link Stage} that should be
 * used to display {@link DatabaseCreatorView} gui-objects.
 */
public class DatabaseCreatorWindow extends Stage {

    private static final double WIDTH = 741;
    private static final double HEIGHT = 400;

    public DatabaseCreatorWindow(DatabaseCreatorView view, Window owner) {
        this.setTitle(getFXMLValues().getString("data.source.adder.window.title"));
        this.setScene(new Scene(view));
        this.initModality(Modality.APPLICATION_MODAL);
        this.setWidth(WIDTH);
        this.setHeight(HEIGHT);
        this.centerOnScreen();
        Theme.applyDefault((Themeable) view);
    }
}

File: src/main/java/com/dansoftware/libraryapp/gui/dbcreator/DatabaseCreatorWindow.kt
package com.dansoftware.libraryapp.gui.dbcreator

import com.dansoftware.libraryapp.gui.util.LibraryAppStage
import javafx.stage.Modality
import javafx.stage.Window

/**
 * A DatabaseCreatorWindow is a javaFX [Stage] that should be
 * used to display [DatabaseCreatorView] gui-objects.
 */
class DatabaseCreatorWindow(view: DatabaseCreatorView, owner: Window?) : LibraryAppStage("window.dbcreator.title", view) {
    init {
        initModality(Modality.APPLICATION_MODAL)
        initOwner(owner)
        width = 741.0
        height = 400.0
        centerOnScreen()
    }
}